cairo_programs/fast_ec_add.cairo

```diff
@@ -0,0 +1,91 @@
+%builtins range_check
+from nondet_bigint3 import nondet_bigint3
+from starkware.cairo.common.cairo_secp.bigint import BigInt3, UnreducedBigInt3
+from starkware.cairo.common.cairo_secp.field import is_zero, unreduced_mul, unreduced_sqr, verify_zero
+from starkware.cairo.common.cairo_secp.ec import EcPoint, compute_slope, ec_double
+
+func fast_ec_add{range_check_ptr}(point0 : EcPoint, point1 : EcPoint) -> (res : EcPoint):
+    # Check whether point0 is the zero point.
+    if point0.x.d0 == 0:
+        if point0.x.d1 == 0:
+            if point0.x.d2 == 0:
+                return (point1)
+            end
+        end
+    end
+
+    # Check whether point1 is the zero point.
+    if point1.x.d0 == 0:
+        if point1.x.d1 == 0:
+            if point1.x.d2 == 0:
+                return (point0)
+            end
+        end
+    end
+
+    let (slope : BigInt3) = compute_slope(point0, point1)
+    let (slope_sqr : UnreducedBigInt3) = unreduced_sqr(slope)
+
+    %{
+        # TEST
+        from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+        slope = pack(ids.slope, PRIME)
+        x0 = pack(ids.point0.x, PRIME)
+        x1 = pack(ids.point1.x, PRIME)
+        y0 = pack(ids.point0.y, PRIME)
+
+        value = new_x = (pow(slope, 2, SECP_P) - x0 - x1) % SECP_P
+    %}
+    let (new_x : BigInt3) = nondet_bigint3()
+
+    %{
+    # TEST 
+    value = new_y = (slope * (x0 - new_x) - y0) % SECP_P 
+    %}
+
+    let (new_y : BigInt3) = nondet_bigint3()
+
+    verify_zero(
+        UnreducedBigInt3(
+        d0=slope_sqr.d0 - new_x.d0 - point0.x.d0 - point1.x.d0,
+        d1=slope_sqr.d1 - new_x.d1 - point0.x.d1 - point1.x.d1,
+        d2=slope_sqr.d2 - new_x.d2 - point0.x.d2 - point1.x.d2),
+    )
+
+    let (x_diff_slope : UnreducedBigInt3) = unreduced_mul(
+        BigInt3(d0=point0.x.d0 - new_x.d0, d1=point0.x.d1 - new_x.d1, d2=point0.x.d2 - new_x.d2),
+        slope,
+    )
+
+    verify_zero(
+        UnreducedBigInt3(
+        d0=x_diff_slope.d0 - point0.y.d0 - new_y.d0,
+        d1=x_diff_slope.d1 - point0.y.d1 - new_y.d1,
+        d2=x_diff_slope.d2 - point0.y.d2 - new_y.d2),
+    )
+
+    return (EcPoint(new_x, new_y))
+end
+
+
+func main{range_check_ptr: felt}():
+
+    let x = BigInt3(1, 5, 10)
+    let y = BigInt3(2, 4, 20)
+
+    let point_a = EcPoint(x, y)
+    let (point_e) = ec_double(EcPoint(BigInt3(156, 6545, 100010), BigInt3(-5336262886225868, 1324, -113834066795297906)))
```

Comment:
Lets avoid using the ec_double( function we are not testing here) to create a variable, just define point_e as` point_e = EcPoint(....)`
You can take the paremeters from `cairo_programs/secp_ec.cairo` in cairo_rs

Reply:
@mmsc2

Reply:
Don't worry about this, it is not really necessary